### ISSUE TITLE
Migrate task instances with variables

### ISSUE THREAD

**Issue by korthout** [camunda/zeebe#15114]
Introduce a new `VariableIntent` `MIGRATED`.

Add an event applier that updates the following properties of a variable:

- `bpmnProcessId`: the BPMN ID of the target process definition
- `processDefinitionVersion`: the version of the target process definition
- `processDefinitionKey`: the key of the target process definition

In the `ProcessInstanceMigration` command processor, add:
- for each variable of a process instance:
  - append `Variable:MIGRATED` event
- for each child element instance of the process:
  - append `Variable:MIGRATED` event for each variable at that scope

Make sure to clear the variable's `value` when appending these events to avoid exceeding the result records batch size.

Add a test case that show that we can migrate a process instance, and use the variables in activated jobs and can update the variables with the Set Variables RPC.

**Alternatively**
Find a way to migrate all variables for a process instance in a single pass. I'm not sure this is possible.

**Out of scope**
Migrate all variables with a single event.

Blocked by:
- #15113

**Comment by sdorokhova** [comment]
Hi @korthout , I don't think that variable has `elementId` field. Is this a copy-paste error?
But variable has `scopeKey` field. Can it change?

**Comment by korthout** [comment]
Good point @sdorokhova, I've removed the `elementId` from the list in the description.

The `scopeKey` won't change. Variables will stay existing at the same scope. We're not planning to move variables to different scopes. That might be needed when we make Sub Process hierarchy changes possible, but even then I doubt we'll move variables. If we change our mind on this for future iterations, we'll let you know in time.
